### PR TITLE
Expose mechanisms to control cache by the configuration

### LINKED ISSUE #1531: Expose RealmHandle + Schema constructor
For GN purposes, we'll need an additional Realm constructor that avoids all caches (including the managed state one).

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes #1531

### CODE REVIEW

File: Realm/Realm/Native/Configuration.cs [hunk 1]
```diff
@@ -62,5 +62,7 @@ internal string Path
 
         internal ShouldCompactCallback should_compact_callback;
         internal IntPtr managed_should_compact_delegate;
+
+        internal bool enable_cache;
```
Comment (fealebenpae) [inline Realm/Realm/Native/Configuration.cs]: This needs to be explicitly marshaled as `UnmanagedType.I1`.

File: Realm/Realm/RealmConfigurationBase.cs [hunk 2]
```diff
@@ -45,6 +45,8 @@ public abstract class RealmConfigurationBase
 
         internal bool Dynamic;
 
+        internal virtual bool EnableCache => true;
```
Comment (fealebenpae) [inline Realm/Realm/RealmConfigurationBase.cs]: I think we're better off with making this a simple field, like `Dynamic, not a virtual property.

Comment (nirinchev) [reply Realm/Realm/RealmConfigurationBase.cs]: Then we'll have to set it which is annoying :P

Comment (fealebenpae) [reply Realm/Realm/RealmConfigurationBase.cs]: Like virtual dispatch is much better :P

Comment (nirinchev) [reply Realm/Realm/RealmConfigurationBase.cs]: I don't see it => it doesn't exist ;) Okay, okay... 😩

**Pull request merged**